build fail real (probably can either delete the test or update to test the doc-level caching)
```
======================================================================
FAIL: test_caching (corehq.apps.userreports.tests.test_expressions.RelatedDocExpressionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_expressions.py", line 749, in test_caching
    self.assertEqual('foo', self.expression(my_doc, EvaluationContext(my_doc, 0)))
AssertionError: 'foo' != None
``

--- a/corehq/apps/userreports/tests/test_expressions.py
+++ b/corehq/apps/userreports/tests/test_expressions.py
@@ -743,13 +743,13 @@ class RelatedDocExpressionTest(SimpleTestCase):
         self.test_simple_lookup()
 
         my_doc = self.database.get('my-id')
+
+        context = EvaluationContext(my_doc, 0)
+        self.assertEqual('foo', self.expression(my_doc, context))
+
+        my_doc = self.database.get('my-id')
         self.database.mock_docs.clear()
-
-        self.assertEqual({}, self.database.mock_docs)
-        self.assertEqual('foo', self.expression(my_doc, EvaluationContext(my_doc, 0)))
-
-        same_expression = ExpressionFactory.from_spec(self.spec)
-        self.assertEqual('foo', same_expression(my_doc, EvaluationContext(my_doc, 0)))
+        self.assertEqual('foo', self.expression(my_doc, context))
 
 
 class RelatedDocExpressionDbTest(TestCase):

